src/base/utils/enzyme.jl

```diff
@@ -0,0 +1,97 @@
+"""
+    protein_dict(model::GeckoModel, opt_model)
+
+Return a dictionary mapping protein concentrations to their ids.
```

Comment:
- this should mention that it is supposed to work on a solved model (maybe an example with FBA+Gecko would be cool)
- it might be much nicer to make this pipeable, just as with `FBA(mdl) |> flux_vector(mdl)`

Reply:
yep, added the piping!

Reply:
piping for both `flux_dict` and `protein_dict` are now there